plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/AppEngineLibraryContainerInitializer.java

```diff
@@ -0,0 +1,82 @@
+package com.google.cloud.tools.eclipse.appengine.libraries;
+
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.RegistryFactory;
+import org.eclipse.jdt.core.ClasspathContainerInitializer;
+import org.eclipse.jdt.core.IClasspathContainer;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.config.LibraryBuilder;
+import com.google.cloud.tools.eclipse.appengine.libraries.config.LibraryBuilder.LibraryBuilderException;
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * {@link ClasspathContainerInitializer} implementation that resolves containers for App Engine libraries.
+ * <p>
+ * The container path is expected to be in the form of
+ * &lt;value of {@link Library#CONTAINER_PATH_PREFIX}&gt;/&lt;library ID&gt;
+ */
+public class AppEngineLibraryContainerInitializer extends ClasspathContainerInitializer {
+
+  public static final String LIBRARIES_EXTENSION_POINT = "com.google.cloud.tools.eclipse.appengine.libraries";
+
+  private Map<String, Library> libraries;
+
+  @VisibleForTesting
```

Comment:
Good catch, this was actually really missing, without it JDT fails to initialize the containers. And also swallows the error, not even a message in the log file (unless the [logging set to verbose](https://github.com/eclipse/eclipse.jdt.core/blob/master/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java#L3478))